guava/src/com/google/common/util/concurrent/Uninterruptibles.java

```diff
@@ -326,7 +326,67 @@ public static boolean tryAcquireUninterruptibly(
     }
   }
 
-  // TODO(user): Add support for waitUninterruptibly.
+  /**
+   * Invokes {@code object.}{@link Object#wait() wait()} uninterruptibly.
+   *
+   * @param object to wait upon.
+   *
+   * @throws IllegalMonitorStateException if the current thread is not
+   *               the owner of the object's monitor.
+   */
+  @GwtIncompatible // concurrency
+  public static void waitUninterruptibly(Object object) {
+    Preconditions.checkNotNull(object);
+    boolean interrupted = false;
+    try {
+      while (true) {
+        try {
+          object.wait();
+          return;
+        } catch (InterruptedException e) {
+          interrupted = true;
```

Comment:
Ok. I too feel that taking a `Supplier` as an argument is too high level. 
Also, `Object.wait()` is too low level to provide and API for. I too moved away from `Object.wait()` to something like `Condition` in my code. 
I don't want to force this into the API unless it adds substantial value. I will let you take the final call on whether we should go forward with this before I proceed further.

On a side note, I am interested to know more about what you said about phasing out the usage of `Supplier`(?) after Java 8. Would be great if you could point to some issue/article etc. regarding this.